src/app/layout/header/header.component.ts

```diff
@@ -77,74 +49,199 @@ export class HeaderComponent implements OnInit, OnDestroy {
+    this.router.navigate([path]);
   }
 
+  private goToExternal(path: string): void {
+    this.logger.log('[HeaderComponent] Switching to external route: ' + path);
+    window.location.href = path;
+  }
 
-  listenToEvents() {
-    this.eventListeners.push(
-      this.route.queryParams.subscribe(params => {
-        this.plannerFollowQueryParams = {};
-        if (Object.keys(params).indexOf('iteration') > -1) {
-          this.plannerFollowQueryParams['iteration'] = params['iteration'];
+  private alreadyInRecent(contextPath: string): boolean {
+    if (this.recentContexts) {
+      this.recentContexts.forEach(thisRecentContext => {
+        if (thisRecentContext.path === contextPath) {
+          return true;
         }
-      })
-    );
+      });
+      return false;
+    } else {
+      // this should not happen
+      this.recentContexts = [] as Context[];
+      return false;
+    }
+  }
+
+  // Event handlers
+
+  private onSelectMenuItem(menuItem: MenuItem): void {
+    this.goTo(menuItem);
   }
 
-  login() {
-    this.broadcaster.broadcast('login');
-    this.loginService.redirectToAuth();
+  private onSelectLogout(): void {
+    this.logger.log('[HeaderComponent] Logging out user.');
+    this.authentication.logout();
+    this.loggedInUser = null as User;
+    this.headerService.clean();
   }
 
-  logout() {
-    this.loginService.logout();
+  private onSelectLogin(): void {
+    this.router.navigate(['/login']);
+  }
+
+  private onSelectAbout(): void {
+    this.logger.log('[HeaderComponent] Showing about modal.');
+    // TODO Integration: this currently opens a modal dialog
+    // This should be part of the header, or at least a common component in a library
+  }
+
+  private onSelectCreateSpace(): void {
+    this.logger.log('[HeaderComponent] Showing create new space.');
+    // TODO Integration: this currently opens a modal dialog
+    // This should either be a common component from a library OR an external link to a platform dialog
+  }
+
+  private onSelectRecentContext(context: Context): void {
+    if (context.space) {
+      this.logger.log('[HeaderComponent] Switching current space to ' + context.space.id);
+      // go to internal route of this context
+      this.router.navigate([context.path]);
+    }
+  }
 
+  private onSelectViewAllSpaces(): void {
+    this.goToInternal('/' + this.loggedInUser.attributes.username + '/_spaces');
   }
 
-  onImgLoad() {
-    this.imgLoaded = true;
+  private onSelectAccountHome(): void {
+    this.goToInternal('/_home');
   }
 
-  resetData(): void {
-    this.imgLoaded = false;
+  private onSelectUserProfile(): void {
+    this.goToInternal('/' + this.loggedInUser.attributes.username);
+  }
+
+  private onFollowedLink(url: string): void {
+    // NOP
+  }
+
+  ngOnInit(): void {
+
+    // logout can also be called by an event from other parts of the app
+    this.broadcaster.on<string>('logout')
+      .subscribe(message => {
```

Comment:
Previously this class implemented ` OnDestroy` and in `ngOnDestroy()` it unsubscribed from events. Now, I see that you're subscribing to events here. Is there any need to unsubscribe with this setup?